mobile/lib/providers/asset.provider.dart

```diff
@@ -78,27 +80,48 @@ class AssetNotifier extends StateNotifier<bool> {
   }
 
   Future<bool> deleteLocalOnlyAssets(
+    BuildContext context,
     Iterable<Asset> deleteAssets, {
     bool onlyBackedUp = false,
   }) async {
     _deleteInProgress = true;
     state = true;
     try {
+      // Filter local assets based on the backup status (`isRemote`).
+      // If `onlyBackedUp` is true, only select assets that are also present on the server (`isRemote`).
+      final assetsToDelete = deleteAssets
+          .where((e) => e.isLocal && (!onlyBackedUp || e.isRemote))
+          .toList();
+
+      if (assetsToDelete.isEmpty) {
+        // No assets to delete, exit early
+        ImmichToast.show(
+          context: context,
+          msg: 'No backed-up assets selected for deletion.',
+          gravity: ToastGravity.BOTTOM,
+        );
```

Comment:
Toast handling can be made in the widget and not the provider. In general, we shouldn't be passing `BuildContext` inside business logic unless we have no other option.